declarative/syncprofilewatcher.cpp

```diff
@@ -139,17 +144,38 @@ Sync::SyncStatus SyncProfileWatcher::syncStatus() const
     return mSyncStatus;
 }
 
-void SyncProfileWatcher::startSync() const
+bool SyncProfileWatcher::synchronizing() const
+{
+    return mSyncStatus < Sync::SYNC_ERROR;
+}
+
+void SyncProfileWatcher::startSync()
 {
     if (mSyncProfile) {
-        mSyncClient.startSync(mSyncProfile->name());
+        const QString profileId = mSyncProfile->name();
+        connect(mSyncClient->requestSync(profileId),
```

Comment:
This has a corner case of SyncProfileWatcher deleted while d-bus call is ongoing. Here it will execute the function while 'this' is already dead. Could be avoided by passing 'this' as context parameter before the function, but then it would still leak memory. 

So perhaps roughly every place where SyncClientInterface is called with a method returning a new instance of QDBusPendingCallWatcher should really reparent the returned watcher to the instance handling the call.

Reply:
Or then SyncClientInterface could automatically add a connection to deleteLater when the calls finish. Either way the api of returning new instances of call watchers is not entirely without problems.

Reply:
I still would like to keep the API returning a call watcher. It avoids a lot of boiler plate to always create a watcher on a pending reply (which is always what you need to do when using the async request API). Having also a pointer as a return value allows not to expose the QDBus header in SyncClientInterface (which is good, I think).

But your remarks are valid, so I've added a `parent` optional argument to all the requests, so the call watcher object can be parented to the object using it and not the SyncClient singleton. If I'm not mistaken, I'm always passing `this` in the context, so there should not be any race conditions where the object is destroyed before the D-Bus call finishes. In addition, now that the request is always parented to the object, its deletion should kill the connection before it can be called.

Reply:
Parenting should be indeed enough to avoid races.

However without it "..., [this], ..." is not as good as "..., this, [this], ...". Former gets executed even if 'this' got destroyed as there it's more or less just a pointer.

Reply:
Ah, thanks once again, I somehow missed the `..., this, [this]...` API. Sure the later makes much more sense in this context. I'll remember it for next time.